packages/carbon-web-components/src/components/file-uploader/file-uploader-button.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { classMap } from 'lit/directives/class-map.js';
+import { LitElement, html } from 'lit';
+import { property, customElement } from 'lit/decorators.js';
+import { prefix } from '../../globals/settings';
+import HostListenerMixin from '../../globals/mixins/host-listener';
+import ifNonEmpty from '../../globals/directives/if-non-empty';
+import styles from './file-uploader.scss';
+import { BUTTON_KIND, BUTTON_SIZE } from '../button/defs';
+
+export { FORM_ELEMENT_COLOR_SCHEME as TILE_COLOR_SCHEME } from '../../globals/shared-enums';
+
+/**
+ * File uploader button .
+ *
+ * @element cds-file-uploader-container
+ * @fires cds-file-uploader-button-changed The custom event fired when there is a user gesture to select files to upload.
+ */
+@customElement(`${prefix}-file-uploader-button`)
+class CDSFileUploaderButton extends HostListenerMixin(LitElement) {
+  /**
+   * Handles `click` event on the button.
+   */
+  private _handleClick(event) {
+    event.target.value = null;
+    const { selectorInput } = this.constructor as typeof CDSFileUploaderButton;
+    this?.shadowRoot?.querySelector(selectorInput)?.setAttribute('value', '');
+    (this?.shadowRoot?.querySelector(selectorInput) as HTMLElement).click();
+  }
+
+  /**
+   * Handles `keydown` event on the button.
+   */
+  private _handleKeyDown(event) {
+    const { selectorInput } = this.constructor as typeof CDSFileUploaderButton;
+    if (event.key === 'Enter' || event.key === 'Space') {
+      this?.shadowRoot?.querySelector(selectorInput)?.setAttribute('value', '');
+      (this?.shadowRoot?.querySelector(selectorInput) as HTMLElement).click();
+    }
+  }
+
+  /**
+   * Handles user gesture to select files to upload.
+   *
+   * @param event The event.
+   */
+  private _handleChange(event: Event | DragEvent) {
+    const addedFiles = this._getFiles(event);
+    const { eventChange, selectorInput } = this
+      .constructor as typeof CDSFileUploaderButton;
+    this.dispatchEvent(
+      new CustomEvent(eventChange, {
+        bubbles: true,
+        composed: true,
+        detail: {
+          addedFiles,
+        },
+      })
+    );
+
+    const fileInput = this?.shadowRoot?.querySelector(selectorInput);
+    if (fileInput) {
+      (fileInput as HTMLInputElement).value = ''; // carbon-web-components#904
```

Comment:
What's this comment mean?

Reply:
oh forgot to come back to that it was part of the original code..... but it has been [fixed](https://github.com/carbon-design-system/carbon-web-components/issues/904), so i'll remove this comment